Apps/Encounter/src/Models/EncounterModel.cs

```diff
@@ -61,9 +63,15 @@ public class EncounterModel
         /// <returns>The newly created Encounter object.</returns>
         public static EncounterModel FromODRClaimModel(Claim model)
         {
+#pragma warning disable CA5351 // Do Not Use Broken Cryptographic Algorithms
+#pragma warning disable SCS0006 // Weak hashing function
+            using var md5CryptoService = MD5.Create();
+#pragma warning restore SCS0006 // Weak hashing function
+#pragma warning restore CA5351 // Do Not Use Broken Cryptographic Algorithms
+
             return new EncounterModel()
             {
-                Id = model.ClaimId.ToString(),
+                Id = new Guid(md5CryptoService.ComputeHash(Encoding.Default.GetBytes($"{model.ServiceDate:yyyyMMdd}{model.SpecialtyDesc}{model.PractitionerName}{model.LocationName}{model.LocationAddress.Province}{model.LocationAddress.City}{model.LocationAddress.PostalCode}{model.LocationAddress.AddrLine1}{model.LocationAddress.AddrLine2}{model.LocationAddress.AddrLine3}{model.LocationAddress.AddrLine4}"))).ToString(),
```

Comment:
nit, for readability I would use a string builder and add each of the qualifiers parameters

Reply:
I prefer a 1 line code instead as you can configure to have wrap mode in VS, but it is fine :-) Updated